logic/resolvable/Concludable.java

```diff
@@ -865,7 +845,7 @@ private static class Extractor {
         private final Set<Concludable> concludables = new HashSet<>();
 
         Extractor(Conjunction conjunction) {
-            assert conjunction.isCoherent();
+            assert conjunction.isCoherent(); // TODO: Confirm we need this, Fix failures this causes.
```

Comment:
This is what causes the failures. We would ideally keep it.
It just is triggered before the rule throws an exception because the when or then are incoherent conjunctions.

I can fix it in a few ways:
1. Make Rule.of validate the coherence of when & then conjunctions before constructing the Rule
2. Delay the concludable extraction till after the rule-validation (done in the .of method)